### PR TITLE
fix: fetch changelog from self-hosted gitlab independent from url

### LINKED ISSUE #6704: Support of "self hosted gitlab" changelog
**What Renovate type are you using?**

self hosted with GitLab Self Hosted CE

**Describe the bug**
Considering this line of code :
https://github.com/renovatebot/renovate/blob/4f48cd1a4e0022e14266b070d433520fb805b277/lib/workers/pr/changelog/index.ts#L28

I think we can't get our changelogs from our self hosted gitlab CE because our url looks like this : `https://git.foo.bar`

**Relevant debug logs**

```
DEBUG: Repository URL does not match any known github hosts - skipping changelog retrieval (repository=mygroup/myproject)
       "manager": "regex",
       "depName": "myDep",
       "sourceUrl": "https://git.foo.bar/mygroup/mydep"
```

it uses the default choice : GitHub

**To Reproduce**

**Additional context**

### PR DESCRIPTION
**Pull request by casdevs**
## Changes:

fetches changelog from self-hosted Gitlab instance even if its url does not contain the word "gitlab"

## Context:

Closes #6704

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

### CODE REVIEW

**Comment by casdevs** [comment]
I'm not sure about 2-3 other source code locations where also only `.includes('gitlab')` is checked.

Are my code changes enough to close #6704?

File: lib/workers/pr/changelog/index.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import URL from 'url';
+
```
Comment (viceice) [inline lib/workers/pr/changelog/index.ts:2]:
```suggestion
```
Use global url object. Module is deprecated.

https://nodejs.org/api/url.html#url_new_url_input_base

File: lib/workers/pr/changelog/index.ts [hunk 2]
```diff
@@ -27,7 +29,12 @@ export async function getChangeLogJSON(
 
     let res: ChangeLogResult | null = null;
 
-    if (args.sourceUrl?.includes('gitlab')) {
+    if (
+      args.sourceUrl?.includes('gitlab') ||
+      (args.platform === 'gitlab' &&
+        URL.parse(args.sourceUrl).hostname ===
+          URL.parse(args.endpoint).hostname)
```
Comment (viceice) [inline lib/workers/pr/changelog/index.ts:36]:
```suggestion
        new URL(args.sourceUrl).hostname ===
          new URL(args.endpoint).hostname)
```

**Review by viceice — CHANGES_REQUESTED** [review]
Needs tests

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**